Python/Product/Analysis/LanguageServer/Server.cs

```diff
@@ -373,8 +373,14 @@ private sealed class ReloadModulesQueueItem : IAnalyzable {
                 return default(T);
             }
             try {
-                var assembly = File.Exists(assemblyName) ? AssemblyName.GetAssemblyName(assemblyName) : new AssemblyName(assemblyName);
-                var type = Assembly.Load(assembly).GetType(typeName, true);
+                Type type;
+                if (Path.IsPathRooted(assemblyName)) {
+                    type = Assembly.LoadFrom(assemblyName).GetType(typeName, true);
```

Comment:
`Assembly.LoadFrom` loads assembly in a different hive which may create problems in the future.
`AssemblyName.GetAssemblyName(assemblyName)` creates `AssemblyName` that references absolute file path.
What issue are you trying to fix?

Reply:
My assembly is in an arbitrary location on disk, and getting the `AssemblyName` from the path works fine:
```
AssemblyName.GetAssemblyName(assemblyName);
{foo, Version=1.0.0.0, Culture=neutral, PublicKeyToken=9f01fad7930233b0}
```
but when I actually try to load it, it cannot be found:
```
Assembly.Load(AssemblyName.GetAssemblyName(assemblyName))
'Assembly.Load(AssemblyName.GetAssemblyName(assemblyName))' threw an exception of type 'System.IO.FileNotFoundException'
```
If there's a way to add search paths to `Assembly.Load` at runtime I'd be open to that, but as far as I can tell that's what `Assembly.LoadFrom` is for.

Reply:
In RTVS we do this
https://github.com/Microsoft/RTVS/blob/ea74954fd2165c30e1e84ff8452373f18d40ab9f/src/VsCode/LanguageServer/Impl/Services/ServiceContainer.cs

```cs

            try {
                var thisAssemblyName = thisAssembly.GetName();
                var name = Path.GetFileNameWithoutExtension(platformAssemblyName);
                var token = thisAssemblyName.GetPublicKeyToken();
                var tokenString = token != null && token.Length > 0 ? Encoding.ASCII.GetString(token) : "null";
                var asmName = new AssemblyName($"{name}, Version={thisAssemblyName.Version}, Culture=neutral, PublicKeyToken={tokenString}");
                assembly = Assembly.Load(asmName);
            } catch(IOException) {
                var thisAssemblyPath = thisAssembly.GetAssemblyPath();
                var assemblyLoc = Path.GetDirectoryName(thisAssemblyPath);
                var platformServicesAssemblyPath = Path.Combine(assemblyLoc, platformAssemblyName);
                assembly = AssemblyLoadContext.Default.LoadFromAssemblyPath(platformServicesAssemblyPath);
            }
```